answerkey/laziness/02.answer.scala

```diff
@@ -3,9 +3,8 @@
 `take` first checks if n==0. In that case we need not look at the stream at all.
 */
 def take(n: Int): Stream[A] = this match {
-  case Cons(h, t) if n > 1 => cons(h(), t().take(n - 1))
-  case Cons(h, _) if n == 1 => cons(h(), empty)
-  case _ => empty
+  case Cons(h, t) if (n > 0) => cons(h(), t().take(n - 1))
```

Comment:
This is a little stricter than we wanted---we wanted `cons(1, sys.error("explode")).take(1)` to return `Stream(1)`, but your simpler implementation would throw the error. This is a design choice---your implementation is not wrong, it just depends on what you think users are expecting, and what you think the better API is. 

We'll probably leave the implementation as is.